Assets/Scripts/DataTypes/Edges/ITrack.cs

```diff
@@ -0,0 +1,12 @@
+using System.Collections.Generic;
+
+namespace DataTypes
+{
+    public interface ITrack
+    {
+        RoadShape shape { get; }
```

Comment:
Sollte man denke ich entfernen, ITrack soll ja nur das interface für autos sein, und die interessiert ja das tatsächliche RoadShape Objekt nicht. Wird auch nirgendwo genutzt.

Reply:
Wird noch nicht genutzt, die Strecken auf den Section werden aber ITrack implementieren.

Reply:
Ich weiß, das ist nicht mein Punkt.
Ich meine nur, dass der Sinn von ITrack (und interfaces generell) ist, dass Car mit allen Objekten klar kommt, die dieses interface implementieren, weil Car nur die im interface spezifizierten Methoden/Properties nutzt.
Also macht es keinen Sinn shape da drin zu haben, da Car ja nur indirekt mit der shape interagiert (Deshalb ist sowas wie GetAbsolutePosition ja nicht in Car implementiert, sondern Edge bzw. der Klasse, die ITrack implementiert).
RoadShape ist quasi nen implementationsdetail, um das sich Car entsprechend nicht kümmern sollte.

Reply:
37c00bc